### PR TITLE
9947 type decl

### PR DESCRIPTION
**Pull request by leviathan747**

### CODE REVIEW

File: doc/notes/9947_type_decl_int.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+tooling for MASL does not parse the body of type definitions on import. This
+simplified the importer, but carries with it limitations. Because the body of
+the type is not parsed into a metamodel, there is no consistent way to determine
+which types reference which other types.
+
+5.3 Model of `model`
+
+There is a model in the `mcshared` project called `model`. This model currently
+consists of a single class with two options that get set by command line flags.
+The class itself is a singleton class and exists only to carry provide command
```
Comment (keithbrown) [inline doc/notes/9947_type_decl_int.md]: grammar error "carry provide"

File: doc/notes/9947_type_decl_int.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+None.
+
+### 9. Code Changes
+
+Fork/Repository: leviathan747/mc
+Branch: 9947_type_decl
+
+<pre>
+
+ bin/xtumlmc_build                                                 |  13 ++++++----
```
Comment (keithbrown) [inline doc/notes/9947_type_decl_int.md]: also need to put new xtumlmc_build.exe inside bin/win/

File: doc/notes/9947_type_decl_int.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+Because `masl` will now default to always emit type forward declarations, MASL
+round trip tests will fail because the input models did not have the forward
+declarations in all cases. This will be solved simply by updating the test
+models to include type forward declarations everywhere.
+
+### 7. Unit Test
+
+7.1 MASL round trip shall pass
+
```
Comment (keithbrown) [inline doc/notes/9947_type_decl_int.md]: Unit test should include eclipse-based test of the original error case

File: doc/notes/9947_type_decl_int.md (hunk 1 above)
Comment (leviathan747) [reply doc/notes/9947_type_decl_int.md]: done

File: doc/notes/9947_type_decl_int.md (hunk 3 above)
Comment (leviathan747) [reply doc/notes/9947_type_decl_int.md]: done. test model attached to issue

File: doc/notes/9947_type_decl_int.md (hunk 2 above)
Comment (leviathan747) [reply doc/notes/9947_type_decl_int.md]: done

**Pull request merged**